src/smoosh/src/smoosh_channel.erl

```diff
@@ -16,27 +16,39 @@
 -include_lib("couch/include/couch_db.hrl").
 
 % public api.
--export([start_link/1, close/1, suspend/1, resume/1, get_status/1]).
--export([enqueue/3, last_updated/2, flush/1]).
+-export([start_link/1, close/1, suspend/1, resume/1, activate/1, get_status/1]).
+-export([enqueue/3, last_updated/2, flush/1, is_key/2, is_activated/1, persist/1]).
 
 % gen_server api.
 -export([
     init/1,
     handle_call/3,
     handle_cast/2,
     handle_info/2,
-    code_change/3,
     terminate/2
 ]).
 
+-define(VSN, 1).
+-define(CHECKPOINT_INTERVAL_IN_MSEC, 180000).
+
+-ifndef(TEST).
+-define(START_DELAY_IN_MSEC, 60000).
+-define(ACTIVATE_DELAY_IN_MSEC, 30000).
+-else.
+-define(START_DELAY_IN_MSEC, 0).
+-define(ACTIVATE_DELAY_IN_MSEC, 0).
+-endif.
+
```

Comment:
Can you add comment to document what various states mean in this module? What `activated = true` or `paused = false` mean?

Reply:
And explanation of various commands we have. What `resume` and `activate` means?

Reply:
Adding FSM diagram here (using http://mermaid-js.github.io/mermaid/#/stateDiagram syntax) would also help.  don't bother with `choice` directives. Just basic transitions with description of a transition.

```
S1 --> S2 : BlaBlaBla
```

Reply:
Added state diagram and comment: 
1. https://github.com/apache/couchdb/pull/3766/commits/5c737f17e96d9a331fef2d6cfa8964b25ee8c53a#diff-2b4c406bce052e564ab66b67f41ba3b2e3c3f85f1d1ca0bc16a08aee2c1e2b72R112
2. https://github.com/apache/couchdb/pull/3766/commits/5c737f17e96d9a331fef2d6cfa8964b25ee8c53a#diff-113cd247cfe56b1aa882a37daa2856fd33bab2e2a6183034eb3a30088960dc13R44-R50